src/main/java/com/parasoft/environmentmanager/jenkins/ExecuteJobBuilder.java

```diff
@@ -182,6 +205,58 @@ public void logMessage(String message) {
 		}
 		return result;
 	}
+	
+	private boolean publishReport(FilePath reportFile, PrintStream logger){
+	    logger.println("Publishing Report to DTP...");
+	    boolean result = true;
+	    EnvironmentManagerPluginDescriptor pluginDescriptor = EnvironmentManagerPlugin.getEnvironmentManagerPluginDescriptor();
+        String dtp = pluginDescriptor.getDtpUrl();
+        String username = pluginDescriptor.getDtpUsername();
+        Secret password = pluginDescriptor.getDtpPassword();
+      
+        Services services = new ServicesImpl(dtp, username, password.getPlainText());
+        String dataCollector = null;
+        try {
+            dataCollector = services.getDataCollectorV2();
+            logger.println("Connecting to DataCollector at: " + dataCollector);
+        } catch (IOException e) {
+           //unable to find datacollector url from services api
+           logger.println("ERROR: Unable to connect to DataCollector.");
+           result = false;
+        }
+        
+        if (result && dataCollector != null) {
+            result = postToDataCollector(reportFile, dataCollector, username, password.getPlainText(), logger);
+            if (result) {
+                logger.println("Successfully published report to DTP.");
+            } else {
+                logger.println("ERROR: unable to publish report to DTP.");
+            }
+        }
+        return result;
+	}
+	
+	private boolean postToDataCollector(FilePath reportFile, String dataCollector, String username, String password, PrintStream logger) {
+	    boolean result = true;
+	    HttpEntity entity = MultipartEntityBuilder.create().setContentType(ContentType.MULTIPART_FORM_DATA)
+                .addPart("file", new FileBody(new File(reportFile.getRemote())))
```

Comment:
It is not safe to use the remove path of the report file on the Jenkins slave because that might be a different machine than the Jenkins server which is executing this code.

Reply:
I verified the new FileCallable implementation with a remote Linux slave node and uploading the report.xml to DTP worked great!